tests/test_k2_user_client_multiloop.py

```
@@ -0,0 +1,130 @@
+
+    class RecordingHandler(BaseHTTPRequestHandler):
+        def do_GET(self) -> None:
+            cid = self.path.removeprefix("/ipfs/")
+            recorded_headers[cid] = dict(self.headers.items())
+            body = b"gateway response"
+            self.send_response(200)
+            self.send_header("Content-Length", str(len(body)))
+            self.end_headers()
+            self.wfile.write(body)
+
+        def log_message(self, format: str, *args: object) -> None:
```

Ruff (0.15.20)

[error] Function argument `format` is shadowing a Python builtin

(A002)